tests/NLog.UnitTests/Targets/DefaultJsonSerializerClassTests.cs

```diff
@@ -0,0 +1,131 @@
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution. 
+// 
+// * Neither the name of Jaroslaw Kowalski nor the names of its 
+//   contributors may be used to endorse or promote products derived from this
+//   software without specific prior written permission. 
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE 
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+// 
+
+using System.Text;
+using NLog.Targets;
+using Xunit;
+
+namespace NLog.UnitTests.Targets
+{
+    /// <summary>
+    ///     Test via <see cref="IJsonConverter" /> path
+    /// </summary>
+    public class DefaultJsonSerializerClassTests : NLogTestBase
+    {
+        private static readonly object _testSyncObject = new object();
+
+        private interface IExcludedInterface
+        {
+        }
+
+        private class ExcludedClass : IExcludedInterface
+        {
+            public string ExcludedString { get; set; }
+            public override string ToString()
+            {
+                return "Skipped"; 
+            }
+        }
+
+        private class IncludedClass
+        {
+            public string IncludedString { get; set; }
+        }
+
+        private class ContainerClass
+        {
+            public string S { get; set; }
+            public ExcludedClass Excluded { get; set; }
+            public IncludedClass Included { get; set; }
+        }
+
+        private static ContainerClass BuildSampleObject()
+        {
+            var testObject = new ContainerClass
+            {
+                S = "sample",
+                Excluded = new ExcludedClass { ExcludedString = "shouldn't be serialized" },
+                Included = new IncludedClass { IncludedString = "serialized" }
+            };
+            return testObject;
+        }
+
+        [Fact]
+        public void IExcludedInterfaceSerializer_RegistersSerializeAsToString_InvokesToString()
+        {
+            try
+            {
+                var testObject = BuildSampleObject();
+
+                var sb = new StringBuilder();
+                var options = new JsonSerializeOptions();
+
+                LogManager.Config().RegisterObjectTransformation<IExcludedInterface>(o => o.ToString());
```

Comment:
So we just need to find a different word than `Config` because it reminds too much of  `Configuration`. And your suggestion is `SerializationConfiguration`. Not sure if it is a better word, since LoggingConfiguration also controls serialization :) But I will try some brainstorming.

Reply:
Renamed `Config()` to `Setup()` and updated code-example in original post.